sqlmesh/core/context_diff.py

```diff
@@ -204,6 +207,13 @@ def promotable_models(self) -> t.Set[str]:
             *self.modified_snapshots,
         } - self.removed
 
+    @property
+    def unpromoted_models(self) -> t.Set[str]:
+        """The set of model names that have have not yet been promoted in the target environment."""
+        return {
+            *self.existing,
```

Comment:
why are you doing this? { *self.existing } this is redundant, just do

return self.existing - self.previsouly_promoted_model_names